packages/timeline-state-resolver/src/integrations/websocketTcpClient/index.ts

```diff
@@ -0,0 +1,154 @@
+import { CommandWithContext, Device, DeviceContextAPI } from '../../service/device'
+import { ActionExecutionResultCode, DeviceStatus, DeviceType, StatusCode, Timeline, TimelineContentTypeWebSocketTcpClient, TSRTimelineContent, WebSocketTCPClientOptions } from 'timeline-state-resolver-types'
+import { WebSocketTcpConnection } from './connection'
+import { WebsocketTcpClientActions } from 'timeline-state-resolver-types'
+
+
+/** this is not an extends but an implementation of the CommandWithContext */
+export interface WebSocketTcpCommand extends CommandWithContext {
+	command: {
+		type: TimelineContentTypeWebSocketTcpClient
+		message: string
+		isBase64Encoded?: boolean
+	}
+	context: string
+  }
+export type WebSocketTcpClientDeviceState = Timeline.TimelineState<TSRTimelineContent>
+
+  export class WebSocketTcpClientDevice extends Device<
+	WebSocketTCPClientOptions,
+	WebSocketTcpClientDeviceState,
+	WebSocketTcpCommand
+> {
+	// Use ! as the connection will be initialized in init:
+	private connection!: WebSocketTcpConnection
+
+	constructor(context: DeviceContextAPI<any>) {
+		super(context)
+	}
+	
+    public async init(options: WebSocketTCPClientOptions): Promise<boolean> {
+		this.connection = new WebSocketTcpConnection(options)
+        await this.connection.connect()
+        return true
+    }
+
+	readonly actions = {
+		[WebsocketTcpClientActions.Reconnect]: async (_id: string) => {
+			await this.connection.connect()
+			return { result: ActionExecutionResultCode.Ok }
+		},
+		[WebsocketTcpClientActions.ResetState]: async (_id: string) => {
+			return { result: ActionExecutionResultCode.Ok }
+		},
+		[WebsocketTcpClientActions.SendWebSocketMessage]: async (_id: string, payload?: Record<string, any>) => {
```

Comment:
The typings for the payload are wrong, as per the `actions.json` it should be
```suggestion
		[WebsocketTcpClientActions.SendWebSocketMessage]: async (_id: string, payload?: SendWebSocketMessagePayload) => {
```

Reply:
Maybe I've misunderstood something, but from looking through the code in the other integrations, it does look like the actions payload need to be a Record<string, xxx> type

So I changed the type from any it to:
`payload?: Record<string, SendWebSocketMessagePayload>`

And the iterate over the Record and send the message directly to the connection.

Reply:
The other integrations are using a loosely typed object which is not ideal but works so long as the implementation is in line with the actions.json file. `payload?: Record<string, SendWebSocketMessagePayload>` however is still not in line with that file nor is your current implementation.

Reply:
has been updated to respect the command structure and use the sendCommand()